commit 6a447b0e3151893f6d4a889956553c06d2e775c6
Merge: f4a2f7866faa d45f89f7437d
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Sun Dec 20 10:44:05 2020 -0800

    Merge tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm
    
    Pull KVM updates from Paolo Bonzini:
     "Much x86 work was pushed out to 5.12, but ARM more than made up for it.
    
      ARM:
       - PSCI relay at EL2 when "protected KVM" is enabled
       - New exception injection code
       - Simplification of AArch32 system register handling
       - Fix PMU accesses when no PMU is enabled
       - Expose CSV3 on non-Meltdown hosts
       - Cache hierarchy discovery fixes
       - PV steal-time cleanups
       - Allow function pointers at EL2
       - Various host EL2 entry cleanups
       - Simplification of the EL2 vector allocation
    
      s390:
       - memcg accouting for s390 specific parts of kvm and gmap
       - selftest for diag318
       - new kvm_stat for when async_pf falls back to sync
    
      x86:
       - Tracepoints for the new pagetable code from 5.10
       - Catch VFIO and KVM irqfd events before userspace
       - Reporting dirty pages to userspace with a ring buffer
       - SEV-ES host support
       - Nested VMX support for wait-for-SIPI activity state
       - New feature flag (AVX512 FP16)
       - New system ioctl to report Hyper-V-compatible paravirtualization features
    
      Generic:
       - Selftest improvements"
    
    * tag 'for-linus' of git://git.kernel.org/pub/scm/virt/kvm/kvm: (171 commits)
      KVM: SVM: fix 32-bit compilation
      KVM: SVM: Add AP_JUMP_TABLE support in prep for AP booting
      KVM: SVM: Provide support to launch and run an SEV-ES guest
      KVM: SVM: Provide an updated VMRUN invocation for SEV-ES guests
      KVM: SVM: Provide support for SEV-ES vCPU loading
      KVM: SVM: Provide support for SEV-ES vCPU creation/loading
      KVM: SVM: Update ASID allocation to support SEV-ES guests
      KVM: SVM: Set the encryption mask for the SVM host save area
      KVM: SVM: Add NMI support for an SEV-ES guest
      KVM: SVM: Guest FPU state save/restore not needed for SEV-ES guest
      KVM: SVM: Do not report support for SMM for an SEV-ES guest
      KVM: x86: Update __get_sregs() / __set_sregs() to support SEV-ES
      KVM: SVM: Add support for CR8 write traps for an SEV-ES guest
      KVM: SVM: Add support for CR4 write traps for an SEV-ES guest
      KVM: SVM: Add support for CR0 write traps for an SEV-ES guest
      KVM: SVM: Add support for EFER write traps for an SEV-ES guest
      KVM: SVM: Support string IO operations for an SEV-ES guest
      KVM: SVM: Support MMIO for an SEV-ES guest
      KVM: SVM: Create trace events for VMGEXIT MSR protocol processing
      KVM: SVM: Create trace events for VMGEXIT processing
      ...

diff --cc arch/x86/kvm/svm/svm.c
index da7eb4aaf44f,941e5251e13f..cce0143a6f80
--- a/arch/x86/kvm/svm/svm.c
+++ b/arch/x86/kvm/svm/svm.c
@@@ -1309,10 -1347,8 +1347,10 @@@ static int svm_create_vcpu(struct kvm_v
  		svm->avic_is_running = true;
  
  	svm->msrpm = svm_vcpu_alloc_msrpm();
 -	if (!svm->msrpm)
 +	if (!svm->msrpm) {
 +		err = -ENOMEM;
- 		goto error_free_vmcb_page;
+ 		goto error_free_vmsa_page;
 +	}
  
  	svm_vcpu_init_msrpm(vcpu, svm->msrpm);